staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -134,3 +137,54 @@ func (q *queue) dump(includeDetails bool) debug.QueueDump {
 		SeatsInUse:        q.seatsInUse,
 	}
 }
+
+// SeatSeconds is a measure of work, in units of seat-seconds, using a fixed-point representation.
+// `SeatSeconds(n)` represents `n/ssScale` seat-seconds.
+// The constants `ssScale` and `ssScaleDigits` are private to the implementation here,
+// no other code should use them.
+type SeatSeconds uint64
+
+// MaxSeatsSeconds is the maximum representable value of SeatSeconds
+const MaxSeatSeconds = SeatSeconds(math.MaxUint64)
+
+// SeatsTimesSeconds produces the SeatSeconds value for the given factors.
+// This is intended only to produce small values, increments in work
+// rather than amount of work done since process start.
+func SeatsTimesSeconds(seats, seconds float64) SeatSeconds {
```

Comment:
The whole point I would like to use int64 (or uint64) instead of float64 is that I would like to get the int-related precision.

I would like to get rid of this function [and leave only the one below - duration is implemented using int...]

Reply:
By "get the int-related precision" do you mean the equivalent of setting `ssScale = 1e9` in this code?  Why is that particularly valuable?

Reply:
I just don't trust floats :)

even "add the same value and substract it may lead to getting different number".

Reply:
I eliminated SeatsTimesSeconds.